### DISCUSSION TITLE
Failure to get go git repository with git-resource

### DISCUSSION THREAD

**Discussion by whpearson** [concourse/concourse#7630] [Help & Support] [answered]
Hi,

We at govuk-paas are having a consistent issue with a git-resource

```
Cloning into '/tmp/build/get/manifests/app-autoscaler/upstream/src/gopkg.in/yaml.v2'...
fatal: unable to access 'https://gopkg.in/yaml.v2/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
fatal: clone of 'https://gopkg.in/yaml.v2' into submodule path '/tmp/build/get/manifests/app-autoscaler/upstream/src/gopkg.in/yaml.v2' failed
```
This has just been failing today and worked fine yesterday at 10am (UTC). No code around this has been changed. Could this be due to https://docs.certifytheweb.com/docs/kb/kb-202109-letsencrypt/ 

This is failing over multiple versions of concourse (7.2.0 and 7.5.0)

Thanks for any help.

**Comment by nekrondev** [comment]
Workaround is currently to add 
```yml
skip_ssl_verification: true
```
to your source node.

**Comment by whpearson** [comment, marked as answer]
Hi, we found a different, workaround. We pushed up a copy of the latest git-resource (1.14.4) to our github repository then configured concourse to use it.

https://github.com/alphagov/paas-cf/commit/178d6bec90e0ea6f9486d4111404b043c481654c

**Reply by nekrondev** [reply]
Yea that's also a good workaroud until someone fixes the official Concourse git resource without compromising security.

**Reply by taylorsilva** [reply]
If you pull the latest image it's probably been fixed already. We re-publish the images daily for all the resource types. The problem here is that Concourse come bundled with the git resource type and thus goes stale.
So another workaround could be to declare the git resource type in your pipelines that use it. Then you'll use the latest published git resource instead of the stale one bundled in your concourse installation.

```yaml
resource_types:
- name: git
  type: registry-image
  source:
    repository: concourse/git-resource
```

**Reply by andrew-edgar** [reply]
This above suggestion usually will not work as most of the workers already have the "git" resource image cached.  I had to change the name to "git2" and change the resources to use "git2".   otherwise we would have to have a way to remove all the cached versions of the git resource in all the running workers.

**Reply by andrew-edgar** [reply]
I have tried this again now and it's failing again.   Is this certificate going to expire regularly?  as then we would need regular updates to the git-resource (2.7.0 is failing for me now).

**Reply by andrew-edgar** [reply]
So I see one issue.

When I add that resource_types I end up getting tag "2.7.0" and not 1.14.0 which is the "supposedly" latest version.   Does dockerhub need to get cleaned up to make sure we get the correct "latest" image?
